src/cli/commands/config.js

```diff
@@ -5,6 +5,41 @@ import type {Reporter} from '../../reporters/index.js';
 import type Config from '../../config.js';
 import buildSubCommands from './_build-sub-commands.js';
 
+const CONFIG_KEYS = [
+  // 'reporter',
+  'rootModuleFolders',
+  'registryFolders',
+  'linkedModules',
+  // 'registries',
+  'cache',
+  'cwd',
+  'looseSemver',
+  'commandName',
+  'preferOffline',
+  'modulesFolder',
+  'globalFolder',
+  'linkFolder',
+  'offline',
+  'binLinks',
+  'ignorePlatform',
+  'ignoreScripts',
+  'disablePrepublish',
+  'nonInteractive',
+  'workspaceRootFolder',
+  'lockfileFolder',
+  'networkConcurrency',
+  'childConcurrency',
+  'networkTimeout',
+  'workspacesEnabled',
+  'pruneOfflineMirror',
+  'enableMetaFolder',
+  'enableLockfileVersions',
+  'linkFileDependencies',
+  'cacheFolder',
+  'tempFolder',
+  'production',
+];
```

Comment:
Couldn't we just iterate over `config`, and possibly blacklist fields if there's internal ones? A whitelist seems a burden to maintain.

Reply:
We can but isn't it safer to have a fixed list? It also makes our job easier by simply using `JSON.stringify` and passing the whitelist. For a blacklist, we need to manually walk over the config object beforehand and copy all to a new object.